powershell-adapter/psDscAdapter/psDscAdapter.psm1

```diff
@@ -49,6 +49,56 @@ function Get-DSCResourceModules
     return $dscModulePsd1List
 }
 
+function Add-AstMembers {
+    param(
+        $AllTypeDefinitions,
+        $TypeAst,
+        $Properties
+    )
+
+    foreach($TypeConstraint in $TypeAst.BaseTypes) {
+        $t = $AllTypeDefinitions | Where-Object {$_.Name -eq $TypeConstraint.TypeName.Name}
+        if ($t) {
+            Add-AstMembers $AllTypeDefinitions $t $Properties
+        }
+    }
+
+    foreach ($member in $TypeAst.Members)
+    {
+        $property = $member -as [System.Management.Automation.Language.PropertyMemberAst]
+        if (($property -eq $null) -or ($property.IsStatic))
+        {
+            continue;
+        }
+        $skipProperty = $true
+        $isKeyProperty = $false
+        foreach($attr in $property.Attributes)
+        {
+            if ($attr.TypeName.Name -eq 'DscProperty')
+            {
+                $skipProperty = $false
+                foreach($attrArg in $attr.NamedArguments)
+                {
+                    if ($attrArg.ArgumentName -eq 'Key')
+                    {
+                        $isKeyProperty = $true
```

Comment:
can't you break early after finding it's a key?

Reply:
yes; added break.